File: micro-common/src/main/java/com/jiangtj/micro/common/exceptions/MicroException.java
package com.jiangtj.micro.common.exceptions;

import lombok.Getter;
import org.springframework.lang.Nullable;

@Getter
public class MicroException extends RuntimeException {

    public MicroException() {
        super();
    }

    public MicroException(String message) {
        super(message);
    }

    public MicroException(String message, @Nullable Throwable cause) {
        super(message, cause);
    }
}

File: micro-common/src/main/java/com/jiangtj/micro/common/exceptions/MicroException.kt
package com.jiangtj.micro.common.exceptions

/**
 * 自定义异常类，继承自 [RuntimeException]，作为项目中自定义异常的基类。
 */
open class MicroException : RuntimeException {
    constructor() : super()

    constructor(message: String) : super(message)

    constructor(message: String, cause: Throwable?) : super(message, cause)
}
